pkg/policy/selectorcache_selector.go

```diff
@@ -104,23 +108,127 @@ func (f *fqdnSelector) remove(dnsProxy identityNotifier) {
 
 // matches returns true if the identity contains at least one label
 // that matches the FQDNSelector's IdentityLabel string
-func (f *fqdnSelector) matches(identity scIdentity) bool {
+func (f *fqdnSelector) matches(_ *slog.Logger, identity scIdentity) bool {
 	return identity.lbls.Intersects(labels.LabelArray{f.selector.IdentityLabel()})
 }
 
 func (f *fqdnSelector) metricsClass() string {
 	return LabelValueSCFQDN
 }
 
+// Version of k8sLbls.requirement where the key is pre-parsed and values are stored as a Set optimal
+// for single-valued values.
+type requirement struct {
+	key      labels.Label       // pre-parsed Label
+	operator selection.Operator // requirement operator
+	values   set.Set[string]    // usually only one value
+}
+
+func (r *requirement) hasValue(value string) bool {
+	return r.values.Has(value)
+}
+
+// Matches returns true if the Requirement matches the input Labels.
+// This is structurally the same as k8sLbls.Requirement.Matches(),
+// but takes LabelArray as an argument instead of a string.
+func (r *requirement) Matches(logger *slog.Logger, ls labels.LabelArray) bool {
```

Comment:
If we're going to implement our own Matches, then we can get rid of dot-delineated labels entirely! They only existed to get labels with colons past the k8s parser.